callout/src/test/java/com/google/apigee/JavaCalloutRemotePolicyExecutionTest.java

```diff
@@ -0,0 +1,201 @@
+package com.google.apigee;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+
+import com.apigee.flow.execution.Action;
+import com.apigee.flow.execution.ExecutionContext;
+import com.apigee.flow.execution.ExecutionResult;
+import com.apigee.flow.message.Message;
+import com.apigee.flow.message.MessageContext;
+import com.google.apigee.Execute.Execution;
+import com.google.protobuf.TextFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class JavaCalloutRemotePolicyExecutionTest {
+  private static final String KEY = "Example";
+  private static final String VALUE = "Value";
+  private static final String URL_PROPERTY_NAME = "urlpropertyname";
+  private static final String SERVER_URL = "url";
+  @Mock MessageContext messageContext;
+  @Mock ExecutionContext executionContext;
+  @Mock RemotePolicyExecutionHandler remotePolicyExecutionHandler;
+  @Mock Message message;
+  private JavaCalloutRemotePolicyExecution callout;
+
+  @Before
+  public void init() {
+    MockitoAnnotations.openMocks(this);
+
+    Map<String, String> properties = new HashMap<>();
+    properties.put(URL_PROPERTY_NAME, SERVER_URL);
+    callout = new JavaCalloutRemotePolicyExecution(properties, remotePolicyExecutionHandler);
+  }
+
+  @Test
+  public void testExecution() throws Exception {
+    Execution.Builder executionBuilder = Execution.newBuilder();
+    TextFormat.merge(
+        "messageContext {"
+            + "  target_request_message {"
+            + "    flow_variables {"
+            + "      key: \""
+            + KEY
+            + "\""
+            + "      value {"
+            + "        flow_variable: \""
+            + VALUE
+            + "\""
+            + "      }"
+            + "    }"
+            + "  }"
+            + "}"
+            + "executionContext {"
+            + "}"
+            + "executionResult {"
+            + "    action: CONTINUE"
+            + "}",
+        executionBuilder);
+    Execution execution = executionBuilder.build();
+
+    ExecutionResult expected = new ExecutionResult(true, Action.CONTINUE);
+    expected.setErrorResponse("");
+    expected.setErrorResponseHeaders(new HashMap<>());
+    expected.setProperties(new Properties());
+
+    doReturn(execution)
+        .when(remotePolicyExecutionHandler)
+        .sendRemoteHttpServerRequest(any(), any());
+    doReturn(message).when(messageContext).getMessage();
+    doNothing().when(message).setContent(anyString());
```

Comment:
usually we try to be as exact as possible .  Here you should be able to match the string?  Try not to use any().  At very least we need the type of the object.  So long as equals() is implemented property, you can match the real object.

Reply:
Do you think you can try to apply this principle in your tests in this PR?  Thanks.